universal/include/userver/formats/universal/common_containers.hpp

```diff
@@ -0,0 +1,296 @@
+#pragma once
+#include <userver/formats/universal/universal.hpp>
+#include <userver/formats/common/items.hpp>
+#include <userver/formats/parse/try_parse.hpp>
+#include <format>
+#include <userver/utils/regex.hpp>
+
+USERVER_NAMESPACE_BEGIN
+namespace formats::universal {
+template <typename T>
+struct Max : public impl::Param<T> {
+  inline constexpr Max(const T& value) :
+      impl::Param<T>(value) {}
+  inline constexpr std::string Check(const T& value) const {
+    return value <= this->value ? "" : this->Error(value);
+  }
+  template <typename Value>
+  inline constexpr std::enable_if_t<meta::kIsRange<Value>, std::string>
+  Check(const Value& value) const {
+    for(const auto& element : value) {
+      if(element > this->value) {
+        return this->Error(element);
+      }
+    }
+    return "";
+  }
+  inline constexpr std::string Error(const T& value) const {
+    return std::format("{} > {}", value, this->value);
+  }
+};
+
+struct MinElements : public impl::Param<std::size_t> {
+  inline constexpr MinElements(const std::size_t& value) :
+      impl::Param<std::size_t>(value) {}
+  template <typename Value>
+  inline constexpr std::string Check(const Value& value) const {
+    if(value.size() >= this->value) {
+      return "";
+    }
+    return this->Error(value);
+  }
+  template <typename Value>
+  inline constexpr auto Error(const Value&) const {
+    return "Error";
+  }
+};
+
+template <typename T>
+struct Min : public impl::Param<T> {
+  inline constexpr Min(const T& value) :
+      impl::Param<T>(value) {}
+  inline constexpr std::string Check(const T& value) const {
+    return value >= this->value ? "" : this->Error(value);
+  };
+  template <typename Value>
+  inline constexpr std::enable_if_t<meta::kIsRange<Value>, std::string>
+  Check(const Value& value) const {
+    for(const auto& element : value) {
+      if(element < this->value) {
+        return this->Error(element);
+      }
+    }
+    return "";
+  }
+  inline constexpr auto Error(const T& value) const {
+    return std::format("{} < {}", value, this->value);
+  };
+};
+template <typename T>
+struct Default : public impl::EmptyCheck, public impl::Param<T> {
+  inline constexpr Default(const T& value) :
+      impl::Param<T>(value) {}
+};
+template <utils::ConstexprString Pattern>
+static const utils::regex kRegex(Pattern);
+
+struct Pattern : public impl::EmptyCheck, public impl::Param<const utils::regex*> {
+  constexpr inline Pattern(const utils::regex& regex) :
+      impl::Param<const utils::regex*>(&regex) {}
+  constexpr inline std::string Check(std::string_view str) const {
+    return utils::regex_match(str, *this->value) ? "" : this->Error(str);
+  }
+  constexpr inline std::string Error(std::string_view) const {
+    return "Error";
+  }
+};
+struct Additional : public impl::EmptyCheck, public impl::Param<bool> {
+  constexpr inline Additional(const bool& value) :
+      impl::Param<bool>(value) {}
+};
+template <>
+struct FieldConfig<int> {
```

Comment:
It should not really be `int`, rather this specialization should be for all `kIsInteger || is_floating_point`

Also, it should not really be `std::vector` and `std::unordered_map`, but `meta::*` stuff, just like constraints of `Parse` from `parse/common_containers.hpp`

`requires`? There is also a `typename Enable` technique from fmt